.github/workflows/build-and-deploy.yml

```diff
@@ -35,42 +35,34 @@ jobs:
           python3 -m pip install --upgrade pip
           pip install -r requirements.txt
 
-      - name: Generate Merged Data
-        run: |
-          python3 gh-data.py
-          python3 merge-data.py
-
-      - name: Switch to `gh-pages` and Preserve Changes
+      - name: Configure Git User
         run: |
           git config --global user.name "github-actions[bot]"
           git config --global user.email "github-actions[bot]@users.noreply.github.com"
 
-          # Stash changes to preserve them
-          git stash --include-untracked
-
-          # Fetch and switch to `gh-pages`
-          git fetch origin gh-pages:gh-pages || git checkout --orphan gh-pages
+      - name: Fetch and Check Out `gh-pages`
+        run: |
+          git fetch origin gh-pages:gh-pages
           git checkout gh-pages
 
-          # Rebase `gh-pages` onto `main` to ensure a linear history
+      - name: Rebase `gh-pages` onto `main` (if triggered by push to main)
+        if: github.event_name == 'push' && github.ref == 'refs/heads/main'
+        run: |
           git rebase main
```

Comment:
I should have said: this is where things get a little whacky.  It's not going to be possible to have a linear history like this.  At some point, you will need to do something.

If you want the history of gh-pages to track main, then you will want merges.  I suggest using `-s theirs` to resolve conflicts on the merge.  That should work.  However, you should probably have some sort of check that the files you are generating do not appear on main.

The other race condition considerations I outlined still apply.

Reply:
Oh, if you wanted to ensure that this gets the right commits, then you need to use `${{ github.sha }}` rather than `main` here.  Otherwise, if there are multiple pushes to main, you could end up with strange things happening.  I think that GitHub's checkout action does funny things with ref names, so maybe that's not a genuine problem, but maybe avoid having to think about that.

Reply:
OK, thanks. Fixed. The generated files are listed in `.gitignore`, do you think that's sufficient?

Reply:
Also fixed `${{ github.sha }}`